pycqed/analysis_v2/Two_state_T1_analysis.py

```diff
@@ -0,0 +1,333 @@
+                'fit_res': self.fit_res['fit_res_P2'],
+                'setlabel': r'P($|f\rangle$) fit',
+                'do_legend': True,
+                'color': 'C2',
+            }
+
+            self.plot_dicts['fit_msg'] = {
+                'plotfn': self.plot_text,
+                'text_string': self.proc_data_dict['fit_msg'],
+                'xpos': 0.1, 'ypos': .9,
+                'ax_id': 'plot_populations_{}'.format(val_name),
+                'horizontalalignment': 'left'}
+
+
+def plot_populations(time, P0, P1, P2, ax,
+                     xlabel='Time', xunit='s',
+                     ylabel='Population', yunit='',
+                     title='', **kw):
+    ax.plot(time, P0, c='C0', linestyle='',
+            label=r'P($|g\rangle$)', marker='v')
+    ax.plot(time, P1, c='C1', linestyle='',
+            label=r'P($|e\rangle$)', marker='^')
+    ax.plot(time, P2, c='C2', linestyle='',
+            label=r'P($|f\rangle$)', marker='d')
+
+    set_xlabel(ax, xlabel, xunit)
+    set_ylabel(ax, ylabel)
+    ax.grid(axis='y')
+    ax.legend()
+    ax.set_ylim(-.05, 1.05)
+    ax.set_title(title)
+
+
+def populations_using_rate_equations(SI: np.array, SX: np.array,
+                                     V0: float, V1: float, V2: float):
+    """
+    Args:
+        SI (array): signal value for signal with I (Identity) added
+        SX (array): signal value for signal with X (π-pulse) added
+        V0 (float):
+        V1 (float):
+        V2 (float):
+    returns:
+        P0 (array): population of the |0> state
+        P1 (array): population of the |1> state
+        P2 (array): population of the |2> state
+        M_inv (2D array) :  Matrix inverse to find populations
+
+    Based on equation (S1) from Asaad & Dickel et al. npj Quant. Info. (2016)
+
+    To quantify leakage, we monitor the populations Pi of the three lowest
+    energy states (i ∈ {0, 1, 2}) and calculate the average
+    values <Pi>. To do this, we calibrate the average signal levels Vi for
+    the transmons in level i, and perform each measurement twice, the second
+    time with an added final π pulse on the 0–1 transition. This final π
+    pulse swaps P0 and P1, leaving P2 unaffected. Under the assumption that
+    higher levels are unpopulated (P0 +P1 +P2 = 1),
+
+     [V0 −V2,   V1 −V2] [P0]  = [S −V2]
+     [V1 −V2,   V0 −V2] [P1]  = [S' −V2]
+
+    where S (S') is the measured signal level without (with) final π pulse.
+    The populations are extracted by matrix inversion.
+    """
+    M = np.array([[V0-V2, V1-V2], [V1-V2, V0-V2]])
+    M_inv = np.linalg.inv(M)
+
+    P0 = np.zeros(len(SI))
+    P1 = np.zeros(len(SX))
+    for i, (sI, sX) in enumerate(zip(SI, SX)):
+        p0, p1 = np.dot(np.array([sI-V2, sX-V2]), M_inv)
+        p0, p1 = np.dot(M_inv, np.array([sI-V2, sX-V2]))
+        P0[i] = p0
+        P1[i] = p1
+
+    P2 = 1 - P0 - P1
+
+    return P0, P1, P2, M_inv
+
+
+def ExpDecayFunclocal(t, tau, amplitude, offset):
```

Comment:
Please look into existing fitting functions, creating a million double "local" functions is bad practice. Of course if your function is different it should be created, but I'm quite sure there are existing functions for this.